executor/simple.go

```diff
@@ -1048,6 +1050,123 @@ func (e *SimpleExec) executeGrantRole(s *ast.GrantRoleStmt) error {
 	return nil
 }
 
+// Should cover same internal mysql.* tables as DROP USER, so this function is very similar
+func (e *SimpleExec) executeRenameUser(s *ast.RenameUserStmt) error {
+
+	failedUsers := make([]string, 0, len(s.UserToUsers))
```

Comment:
From the code below, it seems `failedUsers` will contain at most one user name. Why do you define it as a string?

Reply:
I started by copying the DROP USER case, and changed DELETE with UPDATE basically, and since RENAME USER is a new 'feature' in TiDB, we did not need to consider the difference in compatibility with MySQL 5.7 and MySQL 8.0 (if the command would be atomic or not). So I chose to make it atomic (i.e. rollback and give error on the first failure). I will update the PR, since it no-longer is the exact logic as DROP USER.